bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/discovery/DraytonWiserDiscoveryService.java

```diff
@@ -0,0 +1,201 @@
+import org.openhab.binding.draytonwiser.internal.model.DraytonWiserDTO;
+import org.openhab.binding.draytonwiser.internal.model.HotWaterDTO;
+import org.openhab.binding.draytonwiser.internal.model.RoomDTO;
+import org.openhab.binding.draytonwiser.internal.model.RoomStatDTO;
+import org.openhab.binding.draytonwiser.internal.model.SmartPlugDTO;
+import org.openhab.binding.draytonwiser.internal.model.SmartValveDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DraytonWiserDiscoveryService} is used to discover devices that are connected to a Heat Hub.
+ *
+ * @author Andrew Schofield - Initial contribution
+ */
+@NonNullByDefault
+public class DraytonWiserDiscoveryService extends AbstractDiscoveryService
+        implements DiscoveryService, ThingHandlerService, DraytonWiserRefreshListener {
+
+    private final Logger logger = LoggerFactory.getLogger(DraytonWiserDiscoveryService.class);
+
+    private @Nullable HeatHubHandler bridgeHandler;
+    private @Nullable ThingUID bridgeUID;
+
+    public DraytonWiserDiscoveryService() {
+        super(SUPPORTED_THING_TYPES_UIDS, 30, false);
+    }
+
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    protected void startScan() {
+        final HeatHubHandler handler = bridgeHandler;
+        if (handler != null) {
+            removeOlderResults(getTimestampOfLastScan());
+            handler.setDiscoveryService(this);
+        }
+    }
+
+    @Override
+    public void onRefresh(final DraytonWiserDTO domainDTOProxy) {
+        logger.debug("Received data from Drayton Wise device. Parsing to discover devices.");
+        onControllerAdded(domainDTOProxy);
+        domainDTOProxy.getRooms().forEach(this::onRoomAdded);
+        domainDTOProxy.getRoomStats().forEach(r -> onRoomStatAdded(domainDTOProxy, r));
+        domainDTOProxy.getSmartValves().forEach(sv -> onSmartValveAdded(domainDTOProxy, sv));
+        domainDTOProxy.getHotWater().forEach(hw -> onHotWaterAdded(domainDTOProxy, hw));
+        domainDTOProxy.getSmartPlugs().forEach(sp -> onSmartPlugAdded(domainDTOProxy, sp));
+    }
+
+    private void onControllerAdded(final DraytonWiserDTO domainDTOProxy) {
+        final DeviceDTO device = domainDTOProxy.getExtendedDeviceProperties(0);
+
+        if (device != null) {
+            logger.debug("Controller discovered, model: {}", device.getModelIdentifier());
+            final Map<String, Object> properties = new HashMap<>();
+
+            DraytonWiserPropertyHelper.setControllerProperties(device, properties);
+            final DiscoveryResult discoveryResult = DiscoveryResultBuilder
+                    .create(new ThingUID(THING_TYPE_CONTROLLER, bridgeUID, "controller")).withProperties(properties)
+                    .withBridge(bridgeUID).withLabel("Controller").build();
```

Comment:
Can you add a representation property? Same for below.

Reply:
The controller doesn't have any properties, which might not be good here. I'm not familiar with this device, but it does get the DeviceDTO, which does have a serial number value. So if that makes sense it could be applied here? @andrew-schofield does the controller have a serial number?

Reply:
No :(

Reply:
Resolved in af8d624